modules/app-connector/src/server/connection.ts

```diff
@@ -1,137 +1,196 @@
-    }
+  if (connection.persistFiles) {
+    fs.writeFileSync(connFile, JSON.stringify(connection, null, 2), {encoding: "utf8", flag: "w"});
+    fs.writeFileSync(crtFile, certificateData, {encoding: "utf8", flag: "w"});
+  }
 
-    LOGGER.info("Connected to %s", connection.metadataUrl)
+  if (!jobRenewCertificateStarted && connection!.type === Type.Kyma) {
+    LOGGER.info("Connecting established: certificate renewal job will run at %s", jobRenewCertificate.nextDates()._d);
+    jobRenewCertificate.start();
+  }
 
-    return connection
+  LOGGER.info("Connected to %s", connection.metadataUrl);
+
+  return connection;
 }
 
 export function certificate(): Buffer {
-    if (!established()) {
-        throw new Error("Trying to access connection status without having a connection established. Please call connection.established() upfront to assure an available connection status")
-    }
-    return certificateData
+  if (!established()) {
+    throw new Error(
+      "Trying to access connection status without having a connection established. Please call connection.established() upfront to assure an available connection status"
+    );
+  }
+  return certificateData;
 }
 
 export function privateKey(): Buffer {
-    return privateKeyData
+  return privateKeyData;
 }
 
 export function established(): boolean {
-    return connection != null
+  return connection != null;
 }
 
 function generatePrivateKey(filePath: string): Buffer {
-    LOGGER.debug("Generating new private key: %s", filePath)
-    let keys = forge.pki.rsa.generateKeyPair(2048)
-    const key = forge.pki.privateKeyToPem(keys.privateKey)
-    fs.writeFileSync(filePath, key)
-    LOGGER.info("Generated new private key: %s", filePath)
-    return Buffer.from(key)
+  LOGGER.debug("Generating new private key: %s", filePath);
+  let keys = forge.pki.rsa.generateKeyPair(2048);
+  const key = forge.pki.privateKeyToPem(keys.privateKey);
+  fs.writeFileSync(filePath, key);
+  LOGGER.info("Generated new private key: %s", filePath);
+  return Buffer.from(key);
 }
 
 export function info(): Info {
-    if (!established()) {
-        throw new Error("Trying to access connection status without having a connection established. Please call connection.established() upfront to assure an available connection status")
-    }
-    return connection!;
+  if (!established()) {
+    throw new Error(
+      "Trying to access connection status without having a connection established. Please call connection.established() upfront to assure an available connection status"
+    );
+  }
+  return connection!;
 }
 
 export function destroy(): void {
-    connection = null
+  connection = null;
 
-    if (fs.existsSync(connFile)) {
-        fs.unlinkSync(connFile)
-    }
-    if (fs.existsSync(crtFile)) {
-        fs.unlinkSync(crtFile)
-    }
+  if (fs.existsSync(connFile)) {
+    fs.unlinkSync(connFile);
+  }
+  if (fs.existsSync(crtFile)) {
+    fs.unlinkSync(crtFile);
+  }
+
+  jobRenewCertificate.stop();

```

Comment:
What is with the jobRenewCertificateStarted flag, on re-connect it will be true but job is not running